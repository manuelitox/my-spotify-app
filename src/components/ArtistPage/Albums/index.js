import React, { Component } from 'react'

import Title from 'components/ArtistPage/Title/'
import Album from './Album/'
import PopUp from 'components/generic/PopUp/container'
import Button from 'components/generic/Button/'
import StyledWrapperAlbums from './styles'

class Albums extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    return (
      <div>
        <Title>Albums</Title>
        <StyledWrapperAlbums>
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
        </StyledWrapperAlbums>
        <Button onClick={ this.togglePopUp }>view more albums</Button>
        <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
          <div>
            <h1>Albums list</h1>
            <p>description...</p>
          </div>
        </PopUp>
      </div>      
    )
  } 
}

export default Albums