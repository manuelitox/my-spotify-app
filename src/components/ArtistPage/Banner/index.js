import React from 'react'

import StyledBanner from './styles'
import BannerWrapper from './wrapper'
import StylesWaveArtistIcon from './stylesWaiveIcon'

const Banner = () => (
  <StyledBanner>
    <BannerWrapper />
    <StylesWaveArtistIcon />
  </StyledBanner>
)

export default Banner