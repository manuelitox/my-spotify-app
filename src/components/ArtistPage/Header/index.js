import React from 'react'
import { Link } from 'react-router-dom'

import StyledHeader from './styles'
import SearchFormContainer from 'components/generic/SearchForm/container'
import { ReactComponent as HeadsetIcon } from 'svgs/headset.svg'

const Header = () => (
  <StyledHeader>
    <Link to="/">
      <HeadsetIcon height="45" width="55" viewBox="0 0 70 55"/>
    </Link>
    <SearchFormContainer placeholder="search another artist..." />
  </StyledHeader>
)

export default Header