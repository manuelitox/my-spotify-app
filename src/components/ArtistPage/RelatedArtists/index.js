import React, { Component } from 'react'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Button from 'components/generic/Button/'
import Artist from './Artist/'
import StyledRelatedArtists from './styles'
import StyledWrapperRelatedArtists from './styles-wrapper'
import StyledContainerRelatedArtists from './styles-container'

class RelatedArtists extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    return (
      <StyledRelatedArtists>
        <StyledContainerRelatedArtists>
          <Title color={ theme.blueDark }>Related Artists</Title>
          <StyledWrapperRelatedArtists>
            <Artist id="1" name="Martin Abasto" />
            <Artist id="2" name="Bairam Frootan" />
            <Artist id="3" name="Emmalynn Mazzia" />
            <Artist id="4" name="Gabriel Pires" />
            <Artist id="5" name="Irene Sotelo" />
            <Artist id="6" name="Oea Romana" />
            <Artist id="7" name="Pin Jung-Eym" />
            <Artist id="8" name="Udom Paowsong" />    
            <Artist id="9" name="John Doe" />    
          </StyledWrapperRelatedArtists>
          <Button 
            bg={ theme.blueDark }
            textColor="white"
            onClick={ this.togglePopUp }>
            view related artists
          </Button>
          <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
            <div>
              <h1>Related Artists list</h1>
              <p>description...</p>
            </div>
          </PopUp>          
        </StyledContainerRelatedArtists>
      </StyledRelatedArtists>
    )
  }
}

export default RelatedArtists