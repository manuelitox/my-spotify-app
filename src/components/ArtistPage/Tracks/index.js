import React, { Component } from 'react'

import Track from 'components/generic/Track/'
import Title from 'components/ArtistPage/Title/'
import PopUp from 'components/generic/PopUp/container'
import Button from 'components/generic/Button/'
import StyledWrapperTracks from './styles'

class Tracks extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })
 
  render () {
    return (
      <div>
        <Title>Top Tracks</Title>
        <StyledWrapperTracks>
          <Track name="Empire State of Mind" previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86" releaseDate="2016-07-17" />
          <Track name="No one" previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86" releaseDate="2016-07-17" />
          <Track name="Girl on Fire" previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86" releaseDate="2016-07-17" />
          <Track name="Fallin'" previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86" releaseDate="2016-07-17" />
        </StyledWrapperTracks>
        <Button onClick={ this.togglePopUp }>view more tracks</Button>
        <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
          <div>
            <h1>Top Tracks list</h1>
            <p>description...</p>
          </div>
        </PopUp>        
      </div>
    )
  }
}

export default Tracks