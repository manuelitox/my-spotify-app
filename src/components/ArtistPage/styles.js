import styled from 'styled-components'

const StyledArtistPage = styled.div`
  background-color: ${ props => props.theme.blueDark };
`

export default StyledArtistPage