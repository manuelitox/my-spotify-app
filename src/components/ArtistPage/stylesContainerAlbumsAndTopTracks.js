import styled from 'styled-components'

import WithContainer from 'components/generic/Container/'

const StyledContainerAlbumsAndTopTracks = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  padding-bottom: 40px;
  padding-top: 40px;
  @media only screen and (max-width: ${ props => props.theme.size.medium }) {
    grid-template-columns: 1fr;
    & > div:first-of-type {
      margin-bottom: 30px;
    }
  }
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    padding-top: 0;
  }  
`

export default WithContainer(StyledContainerAlbumsAndTopTracks)