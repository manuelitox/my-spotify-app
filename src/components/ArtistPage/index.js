import React from 'react'
 
import Header from './Header/'
import Banner from './Banner/'
import Albums from './Albums/'
import Tracks from './Tracks/'
import RelatedArtists from './RelatedArtists/'
import StyledArtistPage from './styles'
import StyledContainerAlbumsAndTopTracks from './stylesContainerAlbumsAndTopTracks'

const ArtistPage = () => (
  <StyledArtistPage>
    <Header />
    <Banner />
    <StyledContainerAlbumsAndTopTracks>
      <Albums />
      <Tracks />
    </StyledContainerAlbumsAndTopTracks>
    <RelatedArtists />
  </StyledArtistPage>
)

export default ArtistPage