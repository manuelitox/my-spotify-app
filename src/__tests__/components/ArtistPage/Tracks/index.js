import React from 'react'
import { shallow } from 'enzyme'

import { Tracks } from 'components/ArtistPage/Tracks/'

describe('Component | Tracks', () => {
  const topTracks = {
    tracks: [
      { 
        name: 'song name', 
        preview_url: 'path/to/preview/url', 
        is_playable: true, 
        album: { 
          release_date: '1988' 
        } 
      }
    ]
  }
  let component 
  beforeEach( () => {
    component = shallow(
      <Tracks topTracks={ topTracks } />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})