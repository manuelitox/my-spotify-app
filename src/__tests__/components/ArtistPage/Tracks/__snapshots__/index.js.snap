// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Tracks should render a component 1`] = `
<div>
  <Title>
    Top Tracks
  </Title>
  <styled.div>
    <Track
      name="Empire State of Mind"
      previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
      releaseDate="2016-07-17"
    />
    <Track
      name="No one"
      previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
      releaseDate="2016-07-17"
    />
    <Track
      name="Girl on Fire"
      previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
      releaseDate="2016-07-17"
    />
    <Track
      name="Fallin'"
      previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
      releaseDate="2016-07-17"
    />
  </styled.div>
  <Button
    onClick={[Function]}
  >
    view more tracks
  </Button>
  <OnClickOutside(PopUpContainer)
    eventTypes={
      Array [
        "mousedown",
        "touchstart",
      ]
    }
    excludeScrollbar={false}
    isOpen={false}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    stopPropagation={false}
    togglePopUp={[Function]}
  >
    <div>
      <h1>
        Top Tracks list
      </h1>
      <p>
        description...
      </p>
    </div>
  </OnClickOutside(PopUpContainer)>
</div>
`;
